src/config/mod.rs

```diff
@@ -631,36 +631,58 @@ pub struct TestDefinition<T = OutputId> {
 }
 
 impl TestDefinition<String> {
-    fn resolve_outputs(self, graph: &graph::Graph) -> TestDefinition<OutputId> {
+    fn resolve_outputs(
+        self,
+        graph: &graph::Graph,
+    ) -> Result<TestDefinition<OutputId>, Vec<String>> {
         let TestDefinition {
             name,
             input,
             inputs,
             outputs,
             no_outputs_from,
         } = self;
+        let mut errors = Vec::new();
 
         let output_map = graph.input_map().expect("ambiguous outputs");
 
         let outputs = outputs
             .into_iter()
-            .map(|old| TestOutput {
-                extract_from: output_map.get(&old.extract_from).unwrap().clone(),
-                conditions: old.conditions,
+            .filter_map(|old| {
+                if let Some(output_id) = output_map.get(&old.extract_from) {
+                    Some(TestOutput {
+                        extract_from: output_id.clone(),
+                        conditions: old.conditions,
+                    })
+                } else {
+                    errors.push(format!(r#"Invalid extract_from target in test '{}': transform '{}' does not exist"#, name, old.extract_from));
```

Comment:
Is it definitely a `transform`? Could we just say `input` here?

Reply:
For now, yes since we only support unit testing transforms. Happy to generalize it, though `input` seems a little confusing since it's part of configuring test output assertions. Maybe I can just remove `transform`?

Reply:
Mmm, yeah, that's true. We can just drop `transform` then.